### PR TITLE
pke: update master workflow

### PR DESCRIPTION
**Pull request by bonifaido**
Signed-off-by: Nandor Kracser <bonifaido@gmail.com>

| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | https://github.com/banzaicloud/pipeline-enterprise/pull/123
| License         | Apache 2.0

### What's in this PR?

Add the possibility to update the PKE master node pool.

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)

### CODE REVIEW

**Review by pregnor — APPROVED** [review]

File: internal/providers/pke/pkeworkflow/update_master_node_group_activity.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+	cluster, err := a.clusters.GetCluster(ctx, input.ClusterID)
+	if err != nil {
+		return UpdateMasterNodeGroupActivityOutput{}, err
+	}
+
+	var isMultiMaster bool
+	for _, np := range cluster.GetNodePools() {
+		if input.NodePoolName == np.Name && np.Master {
+			isMultiMaster = np.Count > 1
```
Comment (orymate) [inline internal/providers/pke/pkeworkflow/update_master_node_group_activity.go:118]: do we want to continue with single-master upgrades?

**Review by orymate — APPROVED** [review]

File: internal/providers/pke/pkeworkflow/update_master_node_group_activity.go (hunk 1 above)
Comment (bonifaido) [reply internal/providers/pke/pkeworkflow/update_master_node_group_activity.go:118]: Good question, I haven't tried multi-master TBH, and the current enterprise counterpart of this won't work for sure.

File: internal/providers/pke/pkeworkflow/create_master_activity.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func (a *CreateMasterActivity) Execute(ctx context.Context, input CreateMasterAc
 
 	awsCluster, ok := cluster.(AWSCluster)
 	if !ok {
-		return "", errors.New(fmt.Sprintf("can't create VPC for cluster type %t", cluster))
+		return "", errors.Errorf("can't cast to AWS cluster %t", cluster)
```
Comment (sagikazarmark) [inline internal/providers/pke/pkeworkflow/create_master_activity.go:77]: nit: avoid apostrophes in error messages

**Review by sagikazarmark — APPROVED** [review]

File: internal/providers/pke/pkeworkflow/update_master_node_group_activity.go (hunk 1 above)
Comment (bonifaido) [reply internal/providers/pke/pkeworkflow/update_master_node_group_activity.go:118]: So we probably should prohibit it.

File: internal/providers/pke/pkeworkflow/create_master_activity.go (hunk 2 above)
Comment (bonifaido) [reply internal/providers/pke/pkeworkflow/create_master_activity.go:77]: thanks, fixed

**Pull request merged**